tests/test_docopt.py

```diff
@@ -679,51 +714,271 @@ def test_issue_65_evaluate_argv_when_called_not_when_imported():
+    ("--foo  Description.", [Option(None, "--foo", 0, False)]),
+    ("--foo ARG  Description.", [Option(None, "--foo", 1, None)]),
+    ("--foo ARG  Description. [default: 42]", [Option(None, "--foo", 1, "42")]),
+    # Options can wrap over multiple lines
+    (
+        """\
+         \t --foo ARG, -f ARG  With a long
+
+         wrapped description
+           \t [default: 42]
+         """,
+        [Option("-f", "--foo", 1, "42")],
+    ),
+    # Options can start after whitespace
+    (
+        "\t--foo=<arg>  [default: bar]",
+        [Option(None, "--foo", 1, "bar")],
+    ),
+    (
+        " \t -f ARG, --foo ARG  Description. [default: 42]",
+        [Option("-f", "--foo", 1, "42")],
+    ),
+    # Options can start on the same line as an "options:" heading
+    (
+        "options:-f ARG, --foo ARG  Description. [default: 42]",
+        [Option("-f", "--foo", 1, "42")],
+    ),
+    (
+        "  Special oPtioNs:  --foo ARG  Description. [default: 42]",
+        [Option(None, "--foo", 1, "42")],
+    ),
+    (
+        "  other options: --foo ARG  Description. [default: 42]",
+        [Option(None, "--foo", 1, "42")],
+    ),
+    (
+        """\
+        -a  This is the first option
+
+            -b=<x>  Options don't have to be in an options section
+
+        Options:
+            -c, --charlie  This describes the option.
+            --delta, -d
+                This option has the desc on another line.
+
+        --echo   This option starts after a blank line.
+
+            -f --foxtrot   This option has no comma
+
+        Other Options:
+            -g VAL     This option is after another section heading.
+                       [default: gval]
+        options:-h  This option is on the same line as a heading
+        oPtioNs:--india
+              oPtIons:  -j X
+
+                        [default: jval]
+            and more Options:  --k X  [default: kval]
+        """,
+        [
+            Option("-a", None, 0, False),
+            Option("-b", None, 1, None),
+            Option("-c", "--charlie", 0, False),
+            Option("-d", "--delta", 0, False),
+            Option(None, "--echo", 0, False),
+            Option("-f", "--foxtrot", 0, False),
+            Option("-g", None, 1, "gval"),
+            Option("-h", None, 0, False),
+            Option(None, "--india", 0, False),
+            Option("-j", None, 1, "jval"),
+            Option(None, "--k", 1, "kval"),
+        ],
+    ),
+    # Option with description (or other content) on following line.
+    (
+        """
+        Options:
+            -a
+            -b
+         description of b
+            -c
+        Other Options:
+            -d
+        Other Options:-e
+        """,
+        [
+            Option("-a", None, 0, False),
+            Option("-b", None, 0, False),
+            Option("-c", None, 0, False),
+            Option("-d", None, 0, False),
+            Option("-e", None, 0, False),
+        ],
+    ),
+]
```

Comment:
All of these are the "happy cases" where there truly are options. We should add cases for non-options, such as `some text --foo`

- [ ]

Reply:
Good idea, I've added another case to test this.